lib/appsignal.rb

```diff
@@ -95,26 +109,35 @@ def listen_for_exception(&block)
       raise exception
     end
 
-    def send_exception(exception, tags=nil)
+    def send_exception(exception, tags=nil, namespace=Appsignal::Transaction::HTTP_REQUEST)
       return if !active? || is_ignored_exception?(exception)
-      transaction = Appsignal::Transaction.create(SecureRandom.uuid, ENV.to_hash)
+      unless exception.is_a?(Exception)
+        logger.error('Can\'t send exception, given value is not an exception')
+        return
+      end
+      transaction = Appsignal::Transaction.create(
+        SecureRandom.uuid,
+        namespace,
+        Appsignal::Transaction::GenericRequest.new({})
+      )
       transaction.set_tags(tags) if tags
-      transaction.set_error(exception)
+      transaction.add_exception(exception)
       Appsignal::Transaction.complete_current!
     end
 
     def add_exception(exception)
-      warn '[DEPRECATION] add_exception is deprecated, use set_exception instead'
+      warn '[DEPRECATION] add_exception is deprecated, use set_error instead'
       set_exception(exception)
     end
 
-    def set_exception(exception)
+    def set_error(exception)
```

Comment:
Yes we should make a choice here. In the extension and protocol everything is called error, I think I'll revert back to using exception in the Ruby code because that's more in line with how Ruby works.